sprints/dashboard/tasks.py

```diff
@@ -413,3 +434,146 @@ def unflag_tickets_task() -> None:
+            poker_session = conn.poker_sessions(cell.board_id, state="OPEN", name=session_name)[0]
+
+            # Get IDs of issues belonging only to a specific cell.
+            cell_issue_ids = set(issue.id for issue in issues if issue.key.startswith(cell.key))
+            current_issue_ids = set(conn.poker_session_results(poker_session.sessionId).keys())
+            all_issue_ids = list(current_issue_ids | cell_issue_ids)
+            if all_issue_ids:
+                # User's `name` and `key` are not always the same.
+                members = get_cell_member_names(conn, get_cell_members(conn.quickfilters(cell.board_id)))
+                member_keys = set(user.key for user in all_users if user.displayName in members)
+                current_member_keys = set(user.userKey for user in poker_session.participants)
+                all_member_keys = list(current_member_keys | member_keys)
+
+                if not settings.DEBUG:  # We don't want to trigger this in the dev environment.
+                    poker_session.update(
+                        {
+                            'issuesIds': all_issue_ids,
+                            'participants': all_member_keys,
+                            'scrumMasters': all_member_keys,
+                        },
+                        notify=True,
+                    )
+
+
+@celery_app.task(ignore_result=True)
+def close_estimation_session_task() -> None:
+    """
+    Close all "next-sprint" estimation sessions for every cell.
+
+    The "next-sprint" estimation session needs to match the following criteria:
+    - is open,
+    - has the name matching the result of the `get_next_poker_session_name` function.
+    """
+    with connect_to_jira() as conn:
+        session_name = get_next_poker_session_name(conn)
+
+        for cell in get_cells(conn):
+            poker_sessions = conn.poker_sessions(cell.board_id, state="OPEN", name=session_name)
+            if not settings.DEBUG:  # We really don't want to trigger this in the dev environment.
+                # Handle closing multiple sessions with the same name (though it should not happen).
+                for session in poker_sessions:
+                    conn.close_poker_session(session.sessionId, send_notifications=True)
+
+    move_estimates_to_tickets_task.delay()
+
+
+@celery_app.task(ignore_result=True)
+def move_estimates_to_tickets_task() -> None:
+    """
+    Applies the average vote results from the closed estimation session to the tickets for every cell.
+
+    If there were no votes for a specific issue, its assignee (or another responsible person) is notified.
+    """
+    with connect_to_jira() as conn:
+        session_name = get_next_poker_session_name(conn)
+
+        for cell in get_cells(conn):
+            try:
+                poker_session = conn.poker_sessions(cell.board_id, state="CLOSED", name=session_name)[0]
```

Comment:
Is this guaranteed to return the newest one? Sorting can ensure that :)

Reply:
Good point, though the API response for sessions does not provide any meaningful data we could sort them by. Therefore I've just changed this approach to use all sessions with this name, just like the `close_estimation_session_task` does.